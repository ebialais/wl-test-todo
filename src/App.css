.todo-app {
  text-align: center;
  margin: 0;
  color: #B4F5FF;
  font-family: 'oxygen';
}

