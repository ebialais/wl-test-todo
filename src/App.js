import { useState, useEffect } from 'react';
import './App.css';
import { TodoForm } from './Components/TodoForm';
import { TodoList } from './Components/TodoList';
import { CommunicationLayer } from './Utils/Data';

const com = new CommunicationLayer();

function App() {
  const [todos, setTodos] = useState([]);

  const reload = () => {
    com.fetchTodos(setTodos);
  };

  const addTodos = (action) => {
    com.addTodos(action, reload);
  };

  const toggleCompleted = (id) => {
    com.toggleCompleted(id, reload);
  };

  const deleteTodo = (id) => {
    com.deleteTodo(id, reload)
  };

  useEffect(() => {
    reload();
  }, []);

  return (
    <div className='todo-app'>
      <h1 className='todo-title'>Todo App</h1>
      <TodoForm 
        addTodos={addTodos}
      />
      <TodoList 
        todos={todos}
        toggleCompleted={toggleCompleted}
        className='todo-list'
        deleteTodo={deleteTodo}
      />
    </div>
  );
}

export default App;
