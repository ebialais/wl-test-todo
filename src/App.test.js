import { render, screen } from '@testing-library/react';
import { act } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from './App';

it('renders h1', () => {
  render(<App />);
  const h1 = screen.getByText(/Todo App/);
  expect(h1).toBeInTheDocument();
});

test('render complete All', async () => {
  await act(async () => {
      await act(async () => {
          const renderer = new ShallowRenderer()
          const renderedApp= renderer.render(<App />);
          expect(renderedApp).toMatchSnapshot('rendered App');
      });
  });
});