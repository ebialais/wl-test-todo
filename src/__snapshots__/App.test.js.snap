// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render complete All: rendered App 1`] = `
<div
  className="todo-app"
>
  <h1
    className="todo-title"
  >
    Todo App
  </h1>
  <TodoForm
    addTodo={[Function]}
  />
  <TodoList
    deleteTodo={[Function]}
    todos={Array []}
    toggleCompleted={[Function]}
  />
</div>
`;
